### DISCUSSION TITLE
When is area_of_use None?

### DISCUSSION THREAD

**Discussion by mhoeger** [pyproj4/pyproj#1466] [Q&A] [answered]
Hey all! I'm trying to write code that will check if a given geometry has valid values. The use case is to flag if an incoming vector files for potentially invalid polygons, for example if they have the wrong CRS (or it's a .geojson and we assume it's WGS84 but it's not actually).

The code I wrote is below. However, type checking is upset because `area_of_use` can be None.

My question is: in what scenarios would area_of_use be `None`? How should I the case of `area_of_use` is None (if I need to handle it)?

Any other feedback on this code is also greatly appreciated! I pieced together best I could with context from [this discussion](https://github.com/pyproj4/pyproj/discussions/1294), but may have gotten some things wrong.

```
from shapely.geometry import MultiPolygon, Polygon, mapping
from pyproj import CRS, Transformer


def _is_valid_geometry_for_crs(geometry: Polygon | MultiPolygon, crs: CRS):
    # Get valid extent for CRS (in WGS84)
    bounds = crs.area_of_use.bounds
    # Convert from WGS84 => "native" if neededs
    wgs84 = CRS("EPSG:4326")
    if crs == wgs84:
        minx, miny, maxx, maxy = bounds
    else:
        transformer = Transformer.from_crs(wgs84, crs, always_xy=True)
        minx, miny = transformer.transform(bounds[0], bounds[1])
        maxx, maxy = transformer.transform(bounds[2], bounds[3])

    # ensure that the geometry_bounds is within the valid_bounds
    geometry_bounds = geometry.bounds
    return (
        minx <= geometry_bounds[0]
        and miny <= geometry_bounds[1]
        and maxx >= geometry_bounds[2]
        and maxy >= geometry_bounds[3]
    )
```

**Comment by jjimenezshaw** [comment, marked as answer]
The CRS bounds are not None when they are included in the definition of the CRS (that sounds obvious, but see how you defined the CRS ;). That usually happens only on CRSs coming directly from the EPSG (or ESRI) catalog... and I remember a few cases that were also not defined.
Any other definition of a CRS is usually not including it, like a WKT1 or pipeline coming from a user.

Also being so restrictive with the area of use is not nice. For instance UTM zones are tangent. If you have to work in an area crossed by that division between zones, you have to choose one. (the city of Sevilla in Spain is one example). It is not dramatic to use a CRS a bit out of it bounds. Another problem is when a grid file is involved. There are not data out of the area of the grid file.

**Comment by piyushrpt** [comment]
For example, building a CRS from proj4 strings.

```
crs = CRS.from_proj4("+proj=longlat +a=6378137 +rf=298.257223563 +no_defs")
```
 
Like mentioned above, `area_of_use` has to be a part of the CRS definition.
